data_lang/pretty.py

```diff
@@ -429,6 +460,60 @@ def _Join(self, items, sep, space):
             seq.append(item)
         return _Concat(seq)
 
+    def _Tabular(self, items, sep):
+        # type: (List[MeasuredDoc], str) -> MeasuredDoc
+        """Join `items` together, using one of three styles:
+
+        (showing spaces as underscores for clarity)
+        ```
+        first,_second,_third,_fourth,_fifth,_sixth,_seventh,_eighth
+        ------
+        first,___second,__third,
+        fourth,__fifth,___sixth,
+        seventh,_eighth
+        ------
+        first,
+        second,
+        third,
+        fourth,
+        fifth,
+        sixth,
+        seventh,
+        eighth
+        ```
+
+        The first "single line" style is used if the items fit on one line.  The
+        second "tabular' style is used if the flat width of all items is no
+        greater than `self.max_tabular_width`. The third "multi line" style is
+        used otherwise.
+        """
```

Comment:
Maybe write something about why we're not using a more obvious rule?  It's expensive?

e.g. as long as 2 items fit on every line, it could be tabular, etc.

e.g. if I have a screen of 80, and I have 10 strings of length no more 30, then 2 can fit  on every line


I can sorta see why that requires a search, but it's probably good to document it

Reply:
Done.

That's a good thing to document, really, because the reason for it isn't obvious and explains a bit about the sorts of things you can and can't do with the pretty printing language.

I documented it in a comment (`#`) instead of a doc string (`"""`) b.c. it's about the implementation rather than the interface.